### PR TITLE
Fix submodules

### PR DESCRIPTION
**Pull request by devpow112**
Update various to use `main` over `master`. Update all to latest tagged branch commit. Also fix license checker.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -154,7 +154,6 @@ jobs:
             'llvm:llvm-project/llvm'
           )
           manual_overrides=(
-            'xxhash:c4d6fd586b63b3e25a4ed28a23210791db46df98:only using xxhash'
```
Comment (devpow112) [inline .github/workflows/ci.yml:157]: No longer needed due to change in their license file structure

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -224,7 +223,7 @@ jobs:
             if [ -z "$original_hash" ]; then
               original_license=$(
                 find ./externals/$path -maxdepth 1 -mindepth 1 \
-                -type f -iname 'LICENSE*'
+                -type f -iregex '^\.\/externals\/.+\/license\(\..+\)?$'
```
Comment (devpow112) [inline .github/workflows/ci.yml:226]: Much more exact search for source license.

**Comment by devpow112** [comment]
This will need #412 merged to pass.

**Review by AWoloszyn — DISMISSED** [review]

**Comment by devpow112** [comment]
Looks like latest re2 fails on windows
```
FAILED: externals/CMakeFiles/re2.dir/re2/re2/dfa.cc.obj 
sccache C:\PROGRA~2\MICROS~1\2017\BUILDT~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\cl.exe  /nologo /TP -D_WN_64_BIT -D_WN_X86 -IC:\workspace\source\externals\re2 /DWIN32 /D_WINDOWS   /Z7 /Ob0 /Od /RTC1 -MTd   /wd4204 /D_CRT_SECURE_NO_WARNINGS /D_SCL_SECURE_NO_WARNINGS /D_HAS_EXCEPTIONS=0 /W4 /WX /GR- /MP /D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING /D_SILENCE_CXX17_IS_LITERAL_TYPE_DEPRECATION_WARNING /D_SILENCE_CXX17_RESULT_OF_DEPRECATION_WARNING /D_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING /D_WN_WINDOWS /wd4100 /wd4201 /wd4456 /wd4457 /wd4702 /wd4815 /wd4459 /wd4244 /wd4127 -std:c++17 /showIncludes /Foexternals\CMakeFiles\re2.dir\re2\re2\dfa.cc.obj /Fdexternals\CMakeFiles\re2.dir\re2.pdb /FS -c C:\workspace\source\externals\re2\re2\dfa.cc
C:\workspace\source\externals\re2\re2/stringpiece.h(138): error C2589: '(': illegal token on right side of '::'
C:\workspace\source\externals\re2\re2/stringpiece.h(138): error C2062: type 'unknown-type' unexpected
C:\workspace\source\externals\re2\re2/stringpiece.h(138): error C2059: syntax error: ')'
C:\workspace\source\externals\re2\re2/stringpiece.h(188): error C2589: '(': illegal token on right side of '::'
C:\workspace\source\externals\re2\re2/stringpiece.h(188): error C2062: type 'unknown-type' unexpected
C:\workspace\source\externals\re2\re2/stringpiece.h(188): error C2059: syntax error: ')'
```
Will have to figure out.

**Comment by devpow112** [comment]
I think it's cause they use `std:min`/`std::max` and windows is fucking it up with there macros.

**Pull request merged**